### DISCUSSION TITLE
How to create a D2D effect with Silk?

### DISCUSSION THREAD

**Discussion by myblindy** [dotnet/Silk.NET#1951] [Q&A] [answered]
For effects I need to create a class derived from `public ID2D1EffectImpl, public ID2D1DrawTransform`, but neither is defined as an actual interface, they're both `struct`s. Is there some kind of pattern I can use? Or do I need to rewrite them as plain COM interfaces to derive from?

**Comment by Perksey** [comment, marked as answer]
This is a tricky one which I had hoped noone would pick up on until 3.0! Sadly there's no way to do this today without doing it manually. It is our intention to resolve this in 3.0 by exposing interfaces which can be implemented by C# classes.

**Reply by hez2010** [reply]
Is there any example of doing this manually?
